Makefile.js

```diff
@@ -695,7 +704,16 @@ target.gensite = function(prereleaseVersion) {
 
                 text = `${ruleHeading}${isRecommended ? RECOMMENDED_TEXT : ""}${isFixable ? FIXABLE_TEXT : ""}\n${ruleDocsContent}`;
 
-                text = `---\ntitle: ${ruleName} - Rules\nlayout: doc\n---\n<!-- Note: No pull requests accepted for this file. See README.md in the root directory for details. -->\n\n${text}`;
+                text = [
+                    "---",
+                    `title: ${ruleName} - Rules`,
+                    "layout: doc",
+                    filename.indexOf("rules/") !== -1 && `edit_link: https://github.com/eslint/eslint/edit/master/docs/rules/${baseName}`,
+                    "---",
+                    "<!-- Note: No pull requests accepted for this file. See README.md in the root directory for details. -->",
+                    "",
+                    text
+                ].filter(x => x).join("\n");
```

Comment:
Why is `.filter(x => x)` used here? It seems like this will result in the `""` on line 714 getting removed, which seems redundant.